scl/mariadb/audit.conf

```diff
@@ -0,0 +1,67 @@
+#############################################################################
+# Copyright (c) 2022 Balabit
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 as published
+# by the Free Software Foundation, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# As an additional exemption you are allowed to compile & link against the
+# OpenSSL libraries as published by the OpenSSL project. See the file
+# COPYING for details.
+#
+#############################################################################
+
+# Parse mysql/mariadb audit log
+# 
+# See https://mariadb.com/kb/en/mariadb-audit-plugin/
+# To enable mariadb/mysql audit logging follow instructions
+# Make sure server_audit_output_type is set to 'syslog'
+#
+# Format is [serverhost],[username],[host],[connectionid],[queryid],[operation],[database],[object],[retcode]
+# 
+
+block parser mariadb-audit-parser(prefix('.mariadb.') template("$MESSAGE")) {
+    channel {
+        parser {
+            csv-parser(
+                columns(
+                    '`prefix`serverhost',
+                    '`prefix`username',
+                    '`prefix`host',
+                    '`prefix`connectionid',
+                    '`prefix`queryid',
+                    '`prefix`operation',
+                    '`prefix`database',
+                    '`prefix`object',
+                    '`prefix`retcode'
+                ),
+                delimiters(','),
+                template(`template`),
+                flags(
+                    greedy,
+                    drop-invalid,
+                    escape-backslash
+                ),
+                quote-pairs("''")
+            );
+        };
+        filter {
+            match('^(.*) ([^ ]*)$', value('`prefix`serverhost'), type(pcre), flags(store-matches));
```

Comment:
regexp-parser() allows extraction into name-value pairs directly, this way you wouldn't need to use temporary values like $1, $2

something like:

```
regexp-parser(prefix(`prefix`) patterns("'^(?<syslog_info>.*) (?<serverhost>[^ ]*)$'"))
```

This uses a prefix and named subpatterns (see man pcrepattern, section NAMED SUBPATTERNS) and would do exactly as you wanted to. regexp-parser() also drops non-matching log messages by default.

this avoids having to use $1, $2, etc.